crates/rome_js_analyze/src/analyzers/use_ts_expect_error.rs

```diff
@@ -0,0 +1,233 @@
+use std::ops::Range;
+
+use rome_analyze::{
+    context::RuleContext, declare_rule, ActionCategory, Ast, Rule, RuleCategory, RuleDiagnostic,
+};
+use rome_console::markup;
+use rome_diagnostics::Applicability;
+use rome_js_syntax::{JsAnyStatement, JsSyntaxToken};
+use rome_rowan::{AstNode, AstNodeExt, TriviaPiece};
+
+use crate::JsRuleAction;
+
+declare_rule! {
+    /// Enforces the use of `@ts-expect-error` instead of `@ts-ignore` to get notified when a suppression is no longer necessary.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```ts,expect_diagnostic
+    /// // @ts-ignore
+    /// let foo: boolean = 1;
+    /// ```
+    ///
+    /// ```ts,expect_diagnostic
+    /// // @ts-ignore: Blah blah blah
+    /// let foo: boolean = 1;
+    /// ```
+    ///
+    /// ```ts,expect_diagnostic
+    /// /* @ts-ignore */
+    /// let foo: boolean = 1;
+    /// ```
+    ///
+    /// ```ts,expect_diagnostic
+    /// /** @ts-ignore */
+    /// let foo: boolean = 1;
+    /// ```
+    ///
+    /// ```ts,expect_diagnostic
+    /// /**
+    ///  ** @ts-ignore */
+    /// let foo: boolean = 1;
+    /// ```
+    /// ### Valid
+    ///
+    /// ```ts
+    /// // @ts-expect-error
+    /// let foo: boolean = 1;
+    /// // @ts-expect-error: Blah blah blah
+    /// let foo: boolean = 1;
+    /// /* @ts-expect-error */
+    /// let foo: boolean = 1;
+    /// /** @ts-expect-error */
+    /// let foo: boolean = 1;
+    /// /**
+    /// * @ts-expect-error */
+    /// let foo: boolean = 1;
+    /// /**
+    /// ** @ts-expect-error */
+    /// let foo: boolean = 1;
+    /// ```
+    pub(crate) UseTsExpectError = "useTsExpectError"
+}
+
+const TS_EXPECT_ERROR_SUPPRESSION: &str = "@ts-expect-error";
+
+impl Rule for UseTsExpectError {
+    const CATEGORY: RuleCategory = RuleCategory::Lint;
+
+    type Query = Ast<JsAnyStatement>;
+    /// Because `ts-ignore` could be used in multiple places, we need to track the state of each `@ts-ignore`.
+    /// The first element of tuple is the index of the original leading trivia pieces that need to replace,
+    /// the second element of tuple is a [Vector] of the range of `@ts-ignore` in the original trivia piece text,
+    /// we use a [Vector] to store the range because the `@ts-ignore` could exist multiple time in single trivia piece.
```

Comment:
```suggestion
    /// `ts-ignore` could be used in multiple places inside a comment, we need to track the state of each `@ts-ignore`.
    /// In order to track these comments, we shape the state with a [Vec] of tuples, where a tuple contains:
    /// - the first element is the position  of the original leading trivia that needs to be replaced;
    /// - the second element of the tuple is a [vector](Vec) of the [ranges](Range) of the string `"@ts-ignore"` found in the original trivia piece text;
    /// 
    /// We use a [Vector] because the string `"@ts-ignore"` can be present multiple times inside a comment.
```

1. "track the state" is a bit generic, what is "a state"? Maybe we can say "track the position"